### PR TITLE
Eliminate sys.Ioctl

### PR DESCRIPTION
**Pull request by krader1961**
I started by writing tests for sys.Ioctl but after I had 100% coverage
of that function it occurred to me the function could be eliminated.

Related #1062

### CODE REVIEW

File: pkg/sys/isatty_unix.go [hunk 1]
```diff
@@ -4,12 +4,11 @@ package sys
...
 import (
 	"os"
-	"unsafe"
 )
 
 // IsATTY returns true if the given file is a terminal.
 func IsATTY(file *os.File) bool {
-	var term Termios
-	err := Ioctl(int(file.Fd()), getAttrIOCTL, uintptr(unsafe.Pointer(&term)))
+	//_, err := unix.IoctlGetTermios(int(file.Fd()), getAttrIOCTL)
```
Comment (xiaq) [inline pkg/sys/isatty_unix.go:11]: Do not comment out code, simply remove it.

File: pkg/sys/tc.go [hunk 2]
```diff
@@ -3,20 +3,15 @@
 package sys
 
 import (
-	"unsafe"
-
 	"golang.org/x/sys/unix"
 )
 
+/* This function is currently unused.
```
Comment (xiaq) [inline pkg/sys/tc.go:9]: Remove

File: pkg/sys/termios.go [hunk 3]
```diff
@@ -15,19 +15,16 @@ import (
...
-	var term Termios
-	if err := Ioctl(fd, getAttrIOCTL, uintptr(unsafe.Pointer(&term))); err != nil {
-		return nil, err
-	}
-	return &term, nil
+// TermiosForFd returns a pointer to a Termios structure if the file
+// descriptor is open on a terminal device.
+func TermiosForFd(fd int) (*Termios, error) {
+	term, err := unix.IoctlGetTermios(fd, getAttrIOCTL)
+	return (*Termios)(unsafe.Pointer(term)), err
```
Comment (xiaq) [inline pkg/sys/termios.go:22]: The conversion to unsafe.Pointer is not needed.

File: pkg/sys/isatty_unix.go (hunk 1 above)
Comment (krader1961) [reply pkg/sys/isatty_unix.go:11]: Agreed. That was leftover from when I was dithering on whether to rewrite this function in terms of Go's `unix.IoctlGetTermios` or `TermiosForFd`.

File: pkg/sys/termios.go (hunk 3 above)
Comment (krader1961) [reply pkg/sys/termios.go:22]: Yep, again, leftover from previous changes as I explored various refactorings.

File: pkg/sys/tc.go (hunk 2 above)
Comment (krader1961) [reply pkg/sys/tc.go:9]: That works for me. I left it in case you had a good reason for writing it in the first place.

**Comment by xiaq** [comment]
Unfortunately I had to revert this change since it resulted in "failed to put myself in foreground" errors.

**Pull request merged**